internal/controllers/topology/cluster/patches/variables/value.go

```diff
@@ -31,6 +31,24 @@ const (
 	rightArrayDelim = "]"
 )
 
+type variableNotFoundErr struct {
+	err error
+}
+
+func variableNotFoundError(err error) variableNotFoundErr {
+	return variableNotFoundErr{
+		err: err,
+	}
+}
+
+func (v variableNotFoundErr) Error() string {
+	return v.err.Error()
+}
+
+func (variableNotFoundErr) Cause() error {
```

Comment:
I did this to avoid having to do any checking for substrings etc. to have a common field to check, we could also do a type-switch but this seems more in line with what's done in other errors.

Reply:
Took another look at how it's done with Kubernetes client-go errors (I like the pattern there).

So in general I'm fine doing the comparison that way (in their case it's reason, for us it would be checking cause).

But I would prefer if we can hide that detail behind a util fund like they do.

Essentially instead of `apierrors.IsNotFoudn` we can have `variables.IsNotFoundError()` and do the comparison internally